public/app/features/logs/components/LogRows.tsx

```diff
@@ -56,6 +56,29 @@ interface State {
   renderAll: boolean;
 }
 
+// this class generates react-keys that are guaranteed to be unique.
+// it will try to use the provided `maybeId`, but if that's a duplicate,
+// it will use an index-based key.
+class KeyMaker {
+  seen: Set<string>;
+  count: number;
+  constructor() {
+    this.seen = new Set();
+    this.count = 0;
+  }
+
+  getKey(maybeId: string) {
```

Comment:
Either

```javascript
  getKey(maybeId: string) {
```

or

```javascript
class UniqueKeyMaker {
```

Reply:
sorry, could you explain this in more detail? thanks!

Reply:
Sure! Just edited my comment, I forgot to change the function name.

TL;DR it's to get unique keys so either add unique to the class or to the method.

Reply:
i see. thanks! updated.